dist/datasource.js

```diff
@@ -141,6 +141,18 @@ function (angular, _, dateMath, moment) {
       var dataSource = this;
       var from = dateToMoment(options.range.from, false);
       var to = dateToMoment(options.range.to, true);
+      if(dataSource.periodGranularity=="dashboard" || dataSource.periodGranularity != options.timezone) {
```

Comment:
```suggestion
      if(dataSource.periodGranularity=="dashboard") {
```
I think if the config says Dashboard we should always respect that. But if the config has hardcoded tz then we should respect that